Add instruction of workload cluster backup and restore (#6783)
aws/eks-anywhere: docs/content/en/docs/clustermgmt/cluster-backup-restore/backup-cluster.md

---
title: "Backup cluster"
linkTitle: "Backup cluster"
weight: 20
aliases:
    /docs/tasks/cluster/cluster-backup-restore/backup-cluster/
description: >
  How to backup your EKS Anywhere cluster
---

We strongly advise performing regular cluster backups of all the EKS Anywhere clusters under your management. This ensures that you always have an up-to-date cluster state available for restoration in case the cluster experiences issues or becomes unrecoverable. This document outlines the steps for creating the two essential types of backups required for the [EKS Anywhere cluster restore process]({{< relref "./restore-cluster" >}}).

## Etcd backup

For optimal cluster maintenance, it is crucial to perform regular etcd backups on all your EKS Anywhere management and workload clusters. In the event of a cluster upgrade failure, make sure to **always** execute the etcd backup beforehand. To create an etcd backup for your cluster, follow the guidelines provided in the [External etcd backup and restore]({{< relref "../etcd-backup-restore/etcdbackup" >}}) section.

## Cluster API backup

> Since cluster failures primarily occur following unsuccessful cluster upgrades, EKS Anywhere takes the proactive step of automatically creating backups for the Cluster API objects that capture the states of both the management cluster and its workload clusters. These backups are stored within the management cluster folder, where the upgrade command is initiated from the Admin machine, and are generated before each upgrade process.

Reviewer: Just want to confirm, does the workload cluster upgrade also take a backup of all other clusters including management?
Also, I think adding the folder layout here might be beneficial. Something like this:
```
my-mgmt-cluster/
├── cluster-state-backup-2023-10-11T02_55_56        <------ Folder with a backup of CAPI objects
├── my-mgmt-cluster-eks-a-cluster.kubeconfig
├── my-mgmt-cluster-eks-a-cluster.yaml
└── generated
```

> Although the likelihood of a cluster failure occurring without any associated cluster upgrade operation is relatively low, it is still recommended to manually back up these Cluster API objects on a routine basis. For example, to create a Cluster API backup of a cluster:

Reviewer 2: ```suggestion
Although the likelihood of a cluster failure occurring without any associated cluster upgrade operation is relatively low, it is still recommended to manually back up these Cluster API objects after every upgrade. For example, to create a Cluster API backup of a cluster:
```

Also, we do a backup before every upgrade right, regardless of success or failure? Should we just be using that to backup instead of having the user run this command? My above suggestion is if we change these instructions to backup the restore folder that gets created.

Author: we do for upgrade, but not for create. I'm including manual backup instruction in case a cluster just failed without eksa upgrade.

```bash
MGMT_CLUSTER="mgmt"
MGMT_CLUSTER_KUBECONFIG=${MGMT_CLUSTER}/${MGMT_CLUSTER}-eks-a-cluster.kubeconfig
BACKUP_DIRECTORY=backup-mgmt

# Substitute the container version with whatever EKS Anywhere CLI version you are using

> CONTAINER=public.ecr.aws/eks-anywhere/cli-tools:v0.16.2-eks-a-41

Reviewer: Can we provide a cmd here to set the `CONTAINER`? Most users won't have an easy way of deducing this image.
@abhay-krishna can you provide a fancy cmd for this?

---
title: "Backup cluster"
linkTitle: "Backup cluster"
weight: 20
aliases:
    /docs/tasks/cluster/cluster-backup-restore/backup-cluster/
description: >
  How to backup your EKS Anywhere cluster
---

We strongly advise performing regular cluster backups of all the EKS Anywhere clusters. This ensures that you always have an up-to-date cluster state available for restoration in case the cluster experiences issues or becomes unrecoverable. This document outlines the steps for creating the two essential types of backups required for the [EKS Anywhere cluster restore process]({{< relref "./restore-cluster" >}}).

## Etcd backup

For optimal cluster maintenance, it is crucial to perform regular etcd backups on all your EKS Anywhere management and workload clusters. **Always** take an etcd backup before performing an upgrade so it can be used to restore the cluster to a previous state in the event of a cluster upgrade failure. To create an etcd backup for your cluster, follow the guidelines provided in the [External etcd backup and restore]({{< relref "../etcd-backup-restore/etcdbackup" >}}) section.

## Cluster API backup

> Since cluster failures primarily occur following unsuccessful cluster upgrades, EKS Anywhere takes the proactive step of automatically creating backups for the Cluster API objects that capture the states of both the management cluster and its workload clusters. These backups are stored within the management cluster folder, where the upgrade command is initiated from the Admin machine, and are generated before each management cluster upgrade process. For example, after executing a cluster upgrade command on `mgmt-cluster`, a backup folder is generated with the naming convention of `cluster-state-backup-${timestamp}`:

Reviewer 3: If I'm not mistaken, there is a chance we don't backup all clusters if one of them is not ready and just fallback to only backup the management cluster. Maybe we should point this out?
